### PR TITLE
Add a treetable component

### PR DESCRIPTION
**Pull request by cimpeanvasile**

### CODE REVIEW

File: src/clr-addons/treetable/treetable-row.scss [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+}
+
+clr-treetable .datagrid .datagrid-body clr-tt-row .clr-tt-node-content.datagrid-row {
+  border-top: 1px solid #ddd;
+}
+
+/* Indentation for first cells in rows */
```
Comment (cat-mandres) [inline src/clr-addons/treetable/treetable-row.scss]: write nested CSS for this to make this way shorter

Comment (cimpeanvasile) [reply src/clr-addons/treetable/treetable-row.scss]: good point, thanks for the tip

File: src/clr-addons/treetable/treetable-row.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  onRowClick() {
+    if (this.clickable) {
+      this.toggleExpand();
+    }
+  }
+
+  onCaretClick() {
+    if (!this.clickable) {
```
[the remark this answers is not in the document]
Comment (cimpeanvasile) [reply src/clr-addons/treetable/treetable-row.ts]: Not really. Due to event bubbling, clicking on the caret is also registered as a click on the whole row. Toggle + toggle = no toggle :)

File: src/clr-addons/treetable/treetable.html [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<!--
+  ~ Copyright (c) 2018 Porsche Informatik. All Rights Reserved.
+  ~ This software is released under MIT license.
+  ~ The full license information can be found in LICENSE in the root directory of this project.
+  -->
+
```
[the remark this answers is not in the document]
Comment (cimpeanvasile) [reply src/clr-addons/treetable/treetable.html]: That is not possible, the context of clr-datagrid is needed to instantiate all other datagrid components like columns and cells. If we put it inside the treetable, the cells and columns get instantiated first, without a proper context and the app fails to start.

File: src/clr-addons/treetable/treetable-row.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+export class ClrTreetableRow implements AfterContentInit {
+  expanded = false;
+  hasExpandableRow = false;
+
+  @Input() clickable = false;
+
+  @ContentChildren(ClrTreetableRow) ttRows: QueryList<ClrTreetableRow>;
+
+  ngAfterContentInit(): void {
+    this.hasExpandableRow = this.ttRows.length > 1;
```
[the remark this answers is not in the document]
Comment (cimpeanvasile) [reply src/clr-addons/treetable/treetable-row.ts]: Nope, the host component is also taken into account

File: src/clr-addons/treetable/treetable-row.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Wallabeng) [reply src/clr-addons/treetable/treetable-row.ts]: ok get it :)

File: src/clr-addons/treetable/treetable.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (Wallabeng) [reply src/clr-addons/treetable/treetable.html]: hmm ok, thats bad. i'm just a bit unhappy that it's mixed up like that with datagrid markup. But when we want to use datagrid in background it seems like the way to go :(

**Pull request merged**